perfkitbenchmarker/benchmarks/scimark2_benchmark.py

```diff
@@ -0,0 +1,265 @@
+# Copyright 2015 Zi Shen Lim. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs SciMark2.
+
+Original documentation & code: http://math.nist.gov/scimark2/
+
+SciMark2 is a Java (and C) benchmark for scientific and numerical computing.
+It measures several computational kernels and reports a composite score in
+approximate Mflops (Millions of floating point operations per second).
+
+For convenience, we use the code base at https://github.com/zlim/scimark2,
+which contains both the Java and C versions of SciMark2 in a single
+repository.
+"""
+
+import logging
+import re
+
+from perfkitbenchmarker import regex_util
+from perfkitbenchmarker import sample
+from perfkitbenchmarker import vm_util
+
+SCIMARK2_URL = 'https://github.com/zlim/scimark2/archive/master.zip'
+SCIMARK2_ZIP = '{0}/scimark2-master.zip'.format(vm_util.VM_TMP_DIR)
+SCIMARK2_PATH = '{0}/scimark2-master'.format(vm_util.VM_TMP_DIR)
+
+BENCHMARK_INFO = {'name': 'scimark2',
+                  'description': 'Runs SciMark2',
+                  'scratch_disk': False,
+                  'num_machines': 1}
+
+TEST_PARSE_RESULTS = False
+
+
+def GetInfo():
+  return BENCHMARK_INFO
+
+
+def CheckPrerequisites():
+  pass
+
+
+def Prepare(benchmark_spec):
+  """Install SciMark2 on the target vm.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  vms = benchmark_spec.vms
+  vm = vms[0]
+  logging.info('Preparing SciMark2 on %s', vm)
+  vm.Install('build_tools')
+  vm.Install('wget')
+  vm.InstallPackages('unzip')
+  cmds = [
+      'wget {0} -O {1}'.format(SCIMARK2_URL, SCIMARK2_ZIP),
+      '(cd {0} && rm -rf {1} && unzip {2})'.format(
+          vm_util.VM_TMP_DIR, SCIMARK2_PATH, SCIMARK2_ZIP),
+      '(cd {0} && make build_c)'.format(SCIMARK2_PATH),
+      '(cd {0} && make build_java)'.format(SCIMARK2_PATH)
+  ]
+  for cmd in cmds:
+    vm.RemoteCommand(cmd, should_log=True)
+
+
+def Run(benchmark_spec):
+  """Run SciMark2 on the target vm.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  vms = benchmark_spec.vms
+  vm = vms[0]
+  logging.info('Running SciMark2 on %s', vm)
+  samples = []
+  if TEST_PARSE_RESULTS:
```

Comment:
Adding test code in Run() should not be necessary. Instead, the test case should run ParseResults on canned data and compare output. See https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/tests/benchmarks/unixbench_benchmark_test.py as an example.